rust/core-lib/src/whitespace.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate xi_rope;
+
+use xi_rope::Rope;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum Indentation {
+    Tabs,
+    Spaces(usize)
```

Comment:
nit: I'd give this a trailing `,`, so that if we add a member in the future we get a cleaner diff